store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -8156,6 +8159,34 @@ public void modifyContact(OperationContext octxt, int contactId, ParsedContact p
         return result;
     }
 
+    /**
+     * Modifies the contact's display name in DB and index file
+     * @param octxt
+     * @param listOfContactsToBeModified This list consist list of Map, email as key and Pair of ContactId and 
+     * DisplayName as Value
+     */
+    public void modifyAutoContact(OperationContext octxt, List<Map<String, Pair<Integer, String>>> listOfContactsToBeModified)
+            throws IOException {
+        String email = null;
+        for (Map<String, Pair<Integer, String>> map : listOfContactsToBeModified) {
+            try {
+                for (String key : map.keySet()) {
+                    email = key;
+                    ZimbraLog.mailbox.debug("Modifying contact addr=%s", email);
+                    ParsedContact pc = null;
+                    HashMap<String, String> contactFields = new HashMap<>();
+                    contactFields.put("firstName", map.get(key).getSecond());
+                    contactFields.put("fullName", map.get(key).getSecond());
```

Comment:
`ContactConstants.A_firstName` and `ContactConstants.A_fullName` might be better.

Reply:
done.